### PR TITLE
Remove old spilling tmp files

### PR DESCRIPTION
**Pull request by vladl2802**
### Changelog entry 

Temporary files that are created during spilling now will be stored in /tmp/spilling-service-tmp/node_\<nodeId\>_\<guid\> and will be removed on next SpillingService run.

### Changelog category 

* Improvement

### Additional information

...

### CODE REVIEW

File: ydb/library/yql/dq/actors/spilling/spilling_file.cpp [hunk 1]
```diff
@@ -712,6 +734,55 @@ class TDqLocalFileSpillingService : public TActorBootstrapped<TDqLocalFileSpilli
...
+
+        LOG_I("[RemoveOldTmp] removing at root: " << root);
+
+        static const auto isDirOldTmp = [&nodeIdString, &sessionId](TString dirName) -> bool {
+            // dirName: node_<nodeId>_<sessionId>
+            static constexpr size_t NodeIdBegin = 5;
+            if (dirName.Size() < NodeIdBegin || dirName.substr(0, NodeIdBegin) != "node_") {
+                return false;
+            }  
+            const auto nodeIdEnd = dirName.find('_', NodeIdBegin);
```
Comment (lll-phill-lll) [inline ydb/library/yql/dq/actors/spilling/spilling_file.cpp:751]: nit: this code may be simplified. You can split the string by "_" and check if vector size equals to 3. Then compare the elements of vector with the expected ids

File: ydb/library/yql/dq/actors/spilling/spilling_file.cpp [hunk 2]
```diff
@@ -712,6 +734,55 @@ class TDqLocalFileSpillingService : public TActorBootstrapped<TDqLocalFileSpilli
...
 
+    void HandleWork(TEvPrivate::TEvRemoveOldTmp::TPtr& ev) {
+        const auto& msg = *ev->Get();
+        const auto& root = msg.TmpRoot;
+        const auto nodeIdString = ToString(msg.NodeId);
+        const auto& sessionId = msg.SpillingSessionId;
+
+        LOG_I("[RemoveOldTmp] removing at root: " << root);
+
+        static const auto isDirOldTmp = [&nodeIdString, &sessionId](TString dirName) -> bool {
```
Comment (lll-phill-lll) [inline ydb/library/yql/dq/actors/spilling/spilling_file.cpp:745]: Redundant copy here, please use `const TString& dirName`

File: ydb/library/yql/dq/actors/spilling/spilling_file.cpp [hunk 3]
```diff
@@ -712,6 +734,55 @@ class TDqLocalFileSpillingService : public TActorBootstrapped<TDqLocalFileSpilli
...
+            if (nodeIdEnd == TString::npos || dirName.substr(NodeIdBegin, nodeIdEnd - NodeIdBegin) != nodeIdString) {
+                return false;
+            }
+            if (dirName.substr(nodeIdEnd + 1) == sessionId) {
+                return false;
+            }
+            return true;
+        };
+
+        try {
```
[the remark this answers is not in the document]
Comment (vladl2802) [reply ydb/library/yql/dq/actors/spilling/spilling_file.cpp:761]: TDirIterator can throw on creation if there is no such directory (if it does, maybe we have bigger problems..). TDirIterator during iteration and ForceDelete can also throw if something went wrong. I'm not sure is this try-catch needed, but if something in old tmp removing fails it's not that important to break all the thing.

[the remark this answers is not in the document]
Comment (lll-phill-lll) [reply ydb/library/yql/dq/actors/spilling/spilling_file.cpp:761]: I agree, thank you for the clarification!

File: ydb/library/yql/dq/actors/spilling/spilling_file.cpp [hunk 4]
```diff
@@ -189,6 +195,15 @@ class TDqLocalFileSpillingService : public TActorBootstrapped<TDqLocalFileSpilli
             bool Removed = false;
             TMaybe<TString> Error;
         };
+
+        struct TEvRemoveOldTmp : public TEventLocal<TEvRemoveOldTmp, EvRemoveOldTmp> {
+            TFsPath TmpRoot;
+            ui32 NodeId;
+            TString SpillingSessionId;
+
+            TEvRemoveOldTmp(TFsPath tmpRoot, ui32 nodeId, TString spillingSessionId)
```
[the remark this answers is not in the document]
Comment (vladl2802) [reply ydb/library/yql/dq/actors/spilling/spilling_file.cpp:204]: In order to send this event I need pointer (with new) or THolder (with MakeHolder), but IFAIK they both are requiring some constructor.

[the remark this answers is not in the document]
Comment (lll-phill-lll) [reply ydb/library/yql/dq/actors/spilling/spilling_file.cpp:204]: Yes, you are right, but wouldn't a standard constructor be enough for these purposes? Other events are also created using new of THolder

[the remark this answers is not in the document]
Comment (vladl2802) [reply ydb/library/yql/dq/actors/spilling/spilling_file.cpp:204]: I don't understand how default constructor can be enough. I want to set all fields with local variables anyway.
I looked at several Events and they all are using constructor.

**Review by lll-phill-lll — APPROVED** [review]

**Pull request merged**